### DISCUSSION TITLE
Handling of empty strings

### DISCUSSION THREAD

**Discussion by tanmen** [react-hook-form/react-hook-form#5104] [General] [answered]
It's been a while.
I've always used this site a lot.
I wanted to draft a issue to change some functions.

**Is your feature request related to a problem? Please describe.**
I'm having trouble with a field that I haven't entered being treated as an empty string.

**Describe the solution you'd like**
I want to decide what the initial value should be in the argument of useForm.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by tanmen** [comment]
PS
I'm currently using `setValueAs` to work around this, but it's a pain to apply the above to all items, so I'm hoping I can use use useForm to set it all at once.

As for whether or not to apply the above to items that have already been changed, I don't think it matters either way, but applying it to values that have already been changed as well may be out of scope for your needs.

**Comment by bluebill1049** [comment, marked as answer]
you can write a `validate` function and reuse it.

**Reply by tanmen** [reply]
I couldn't quite figure it out.
Does it look like the following?

```
<form onSubmit={handleSubmit(value =>  onSubmit(validate()))}>
...inputs...
</form>
```

---

validate performs validation and conversion...?

**Reply by bluebill1049** [reply]
on the input level.

```
<input {...register('test', { validate: checkEmpty })} />
```

**Reply by tanmen** [reply]
I figured it out.
However, I am wondering if it can be used to define the unentered state as null.

**Reply by bluebill1049** [reply]
native input will always return empty string

**Reply by tanmen** [reply]
It seems so.
The only problem is that it is very difficult to tell whether something has already been entered and erased, or not.

I suggested this because I felt that you would want it to return only the changed values.

**Reply by tanmen** [reply]
Was my suggestion too subtle?
I thought it would be better to do so, so I suggested it, but if for some reason it was difficult or didn't suit your ideas, you can dismiss it.

**Reply by bluebill1049** [reply]
sorry, I got too many notifications and just saw your reply now. I understand what you mean, that's why people start doing `null` and `undefined`, however it has its own problems as well, but it's going against the HTML standard and often I found it bring more problems than benefits. so I would suggest sticking with HTML standard, if you want to know it's been entered or erased, I would suggest following other form states, such as `touched` and `dirty`.

**Reply by tanmen** [reply]
I understand.
Thank you for sticking with me to help me understand the current situation!
